Support deleting collections

awead opened the pull request:
Adds a delete button to the collection resource page which deletes the collection with a confirmation dialog. Collections are now deleted from Solr synchronously. We may want to revisit this later and adopt the pattern similar to updating indexes via a source method that can be either async or sync, but that is out of scope at this moment.

Fixes #774

mtribone commented:
![screen shot 2017-04-24 at 2 11 26 pm](https://cloud.githubusercontent.com/assets/4163828/25353756/5efb8fba-28fe-11e7-8bfb-706df224e16b.png)

mtribone commented:
![screen shot 2017-04-24 at 2 10 54 pm](https://cloud.githubusercontent.com/assets/4163828/25353763/6686e2ac-28fe-11e7-869e-350a2b54a893.png)

awead commented on app/views/collections/new.html.erb:
```diff
@@ -0,0 +1,3 @@
+<h1>Create New Collection</h1>
+<p class="instructions"><%= t("sufia.batch_uploads.files.concepts") %></p>
```
@mtribone shouldn't this have it's own entry? The text for `sufia.batch_uploads.files.concepts` doesn't seem appropriate for collections.

mtribone replied:
The text is exactly the same. If I give it its own entry we'll need to change it twice. Perhaps a better thing would be to pull it out of batch_uploads and put it in a more generic place?

awead replied:
@mtribone Maybe I'm misunderstanding, but here's what I figure that page will show:

# Create New Collection
A work consists of one or more files that work together as a functional unit [...]

I thought the idea was to explain what collections are. Why are we describing what a work is if we're creating collections?

awead replied:
duh. sorry, screenshot is above. So, yeah, all I'm saying is it looks odd to be talking about works when you're on a collections creation page.

mtribone replied:
Yeah I can see that point. I was asked today to add the same help text for consistency.

mtribone commented:
New collection help text


![screen shot 2017-04-24 at 4 10 01 pm](https://cloud.githubusercontent.com/assets/4163828/25356571/a7fe9090-2908-11e7-8fe6-e51e0190dcd0.png)

mtribone commented:
![screen shot 2017-04-25 at 9 42 38 am](https://cloud.githubusercontent.com/assets/4163828/25388332/9befe14e-299b-11e7-87e0-946346282568.png)

awead committed:
Support deleting collections

Adds a delete button to the collection resource page which deletes the
collection with a confirmation dialog. Collections are now deleted from
Solr synchronously. We may want to revisit this later and adopt the
pattern similar to updating indexes via a source method that can be
either async or sync, but that is out of scope at this moment.
